src/isar/services/utilities/threaded_request.py

```diff
@@ -11,11 +11,12 @@ def __init__(self, request_func: Any) -> None:
         self._exception: Optional[Exception] = None
         self._exception_lock: Lock = Lock()
 
-    def start_thread(self, *request_args) -> bool:
+    def start_thread(self, *request_args, thread_name: str) -> bool:
         if self._is_thread_alive():
             return False
         self._output = None
         self._thread = Thread(target=self._thread_func, args=request_args)
+        self._thread.name = thread_name
```

Comment:
We could make the thread name optional with:
```suggestion
    def start_thread(self, *request_args, **kwargs) -> bool:
        if self._is_thread_alive():
            return False
        self._output = None
        self._thread = Thread(target=self._thread_func, args=request_args, **kwargs)
        self._thread.name = thread_name
```
Then in places we call this function you can simply do:
```
get_inspections_thread = ThreadedRequest(self._process_finished_step)
get_inspections_thread.start_thread(
    self.state_machine.current_step, name="get_inspections"
)
```
And this will get passed over to the creation of the Thread